src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.IncrementalAnalyzerProcessor.cs

```diff
@@ -182,30 +190,29 @@ private void ResetLogAggregator()
                     Func<IIncrementalAnalyzer, T, CancellationToken, Task> runnerAsync,
                     CancellationToken cancellationToken)
                 {
-                    // this is a best effort progress report. since we don't clear up
-                    // reported progress when work is done or cancelled. it is possible
-                    // that last reported work is left in report even if it is already processed.
-                    // but when everything is finished, report should get cleared
-                    _ = _registration.ProgressReporter.Update(GetFilePath(value));
-
-                    foreach (var analyzer in analyzers)
+                    using var evaluating = _registration.ProgressReporter.Evaluating();
                     {
-                        if (cancellationToken.IsCancellationRequested)
-                        {
-                            return;
-                        }
+                        ReportPendingWorkItemCount();
```

Comment:
not certain i'm getting the threadsafety here.  It seems to be grabbing a number of arbitrary work items (which could be changing out from underneath it).  BUt then it will get a set of events as it changes.  But i dont' see how the two values are ensured that htey're in sync.

That said, is the design ehre that that's not a problem because the number doesn't need to be accurate, and really as long as we show things descreasing, and eventually get to a 'stopped' state in the UI that we'll be ok?  If so, it would be good to doc that.

Reply:
sure.